packages/vike-react-query/README.md

```diff
@@ -54,13 +67,15 @@ const Movie = ({ id }: { id: string }) => {
 
 ## `withFallback()`
 
-Using `withFallback`, you can create reusable and independent components, that leverage React 18's suspense streaming feature. (Similar to [Next.js Loading UI and Streaming](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming), but on component level.)
-
-While the query is loading, it renders the `Loading` component.
+Using `withFallback()`, you can define a loading and/or an error fallback compnent:
+ - While the query is loading, the `Loading` component is rendered.
+ - When the query is completed and the data is available, the main component is rendered.
+ - If there is an error during loading or rendering the main component, the `Error` component is rendered instead.
 
-When the query completed and the data is available, the component independently becomes interactive.
+> [!NOTE]  
+> Upon SSR, the main component is directly rendered to HTML (without using `Loading`) and the main component is merely phydrated and the data is re-used (instead of fetching the data twice).
+> The `Loading` component is used only on the client-side (e.g. upon client-side navigation).
```

Comment:
The `Loading` component is rendered to the HTML, then the main component is rendered in the same HTML stream, replacing the `Loading` component even before javascript is downloaded.

Reply:
Ah, of course, otherwise progressive hydration wouldn't work. Thanks.